services/sharepoint-connector/src/msgraph/file-filter.service.ts

```diff
@@ -1,26 +1,23 @@
-import type { DriveItem } from '@microsoft/microsoft-graph-types';
 import { Injectable } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { Config } from '../config';
-
-type DefinedFileProperties =
-  | 'file'
-  | 'id'
-  | 'name'
-  | 'size'
-  | 'webUrl'
-  | 'listItem'
-  | 'lastModifiedDateTime';
-
-type DriveItemWithDefinedProperties = Omit<DriveItem, DefinedFileProperties> & {
-  [key in DefinedFileProperties]: Exclude<DriveItem[key], null | undefined>;
-};
+import { DriveItem, ListItem } from './types/sharepoint.types';
 
 @Injectable()
 export class FileFilterService {
   public constructor(private readonly configService: ConfigService<Config, true>) {}
 
-  public isFileValidForIngestion(item: DriveItem): item is DriveItemWithDefinedProperties {
+  public isListItemValidForIngestion(fields: ListItem['fields']) {
+    return Boolean(
+      fields.FileLeafRef &&
+        typeof fields.FileLeafRef === 'string' &&
+        fields.FileLeafRef.toLowerCase().endsWith('.aspx') &&
```

Comment:
Can `FileLeafRef` be anything but string? o.O What else can it be? If it's just a string we can simplify these check with optional chaining 🙂

```suggestion
    fields.FileLeafRef?.toLowerCase().endsWith('.aspx')
```